src/cargo/core/features.rs

```diff
@@ -411,7 +430,9 @@ impl CliUnstable {
             Ok(true)
         };
 
-        match k {
+        // Permit dashes or underscores in parsing these
+        let normalized_key = k.replace("_", "-");
```

Comment:
I think it's probably best to not do this normalization to canonicalize the keys specified?

Reply:
Ack — removed.